### PR TITLE
Added command completion feature to Dapr CLI

### PR DESCRIPTION
**Pull request by mittz**
# Description

For #197, added a feature to support "auto-completion" for Dapr CLI using the Tab key. In addition to that, updated the README.md to introduce the subcommand `dapr completion`.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #197 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Extended the documentation

I didn't come up with an idea to make this test code, so I checked the [kubectl repo](https://github.com/kubernetes/kubectl/tree/master/pkg/cmd/completion), but it looks that it doesn't test for the completion part.

### CODE REVIEW

**Comment by yaron2** [comment]
What a great addition to the CLI!

**Comment by youngbupark** [comment]
@yaron2 @orizohar Can we bring this to 0.9.0? this is cool!

cc/ @msfussell @amanbha @mukundansundar

File: README.md [hunk 1]
```diff
@@ -375,6 +375,16 @@ You can run Dapr's sidecar only (`daprd`) by omitting the application's command
 $ dapr run --app-id myapp --port 3005 --grpc-port 50001
 ```
 
+### Generate shell completion scripts
```
Comment (youngbupark) [inline README.md:378]: Is there a way to support completion in windows? otherwise, we may need to add statement.

Comment (mittz) [reply README.md:378]: @youngbupark I found that the underlying library, Cobra, supports autocompletion for powershell, so I'll try to update the code to support powershell as well. I'll let you know once I finish the work.

Comment (mittz) [reply README.md:378]: I updated the autocompletion feature for windows powershell as well as bash and zsh. Please let me know if there is anything I could improve.

**Comment by youngbupark** [comment]
Thanks @mittz for your contribution! We will take this to 0.9.0 release.

**Pull request merged**